drivers/include/net/netdev/layer.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2017 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_netdev_api
+ * @{
+ *
+ * @file
+ * @brief       Netdev layer helper functions.
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ */
+
+#ifndef NET_NETDEV_LAYER_H
+#define NET_NETDEV_LAYER_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "net/netdev.h"
+#include "assert.h"
+
+/**
+ * @brief   Add a netdev layer to the netdev layer stack
+ *
+ * @param[in] top     Last netdev device added to the stack.
+ * @param[in] dev     New layer to push to the top of the stack.
+ *
+ * @return  The new top netdev layer of the netdev stack.
+ */
+static inline netdev_t *netdev_add_layer(netdev_t *top, netdev_t *dev)
+{
+    assert(top != NULL);
```

Comment:
This assert could also check for ```dev != NULL```.

Reply:
I can't think of a reason why it should ever be `NULL`, so I've added it.

Reply:
how about ```assert(top && dev);```?

Reply:
I think I prefer to keep them splitted, as it makes it slightly easier to determine which case triggered an assertion (`dev == NULL` or `top == NULL`). I can of course rewrite it to
```
assert(top);
assert(dev);
```

Reply:
ok